hail/src/main/scala/is/hail/expr/ir/BinaryOp.scala

```diff
@@ -37,7 +37,7 @@ object BinaryOp {
       case (_: TInt32, _: TInt32) =>
         val ll = coerce[Int](l)
         val rr = coerce[Int](r)
-        op match {
+        (op: @unchecked) match {
```

Comment:
does the `_` not preclude the need for this?

Reply:
oh, yeah, all over here. There was only one that didn't have a base case, I added `incompatible`